src/urlschemehandler.cpp

```diff
@@ -65,12 +65,17 @@ UrlSchemeHandler::handleMetaRequest(QWebEngineUrlRequestJob* request)
     auto library = KiwixApp::instance()->getLibrary();
     auto book = library->getBookById(zimId);
     if (metaName == "favicon") {
-        std::string content= book.getFavicon();
-        std::string mimeType = book.getFaviconMimeType();
-        QBuffer* buffer = new QBuffer;
-        buffer->setData(content.data(), content.size());
-        connect(request, &QObject::destroyed, buffer, &QObject::deleteLater);
-        request->reply(QByteArray::fromStdString(mimeType), buffer);
+        try {
+          auto library = KiwixApp::instance()->getLibrary();
+          auto book = library->getBookById(zimId);
```

Comment:
The variables "library" and "book" are already set line 65 and 66 aren't they ?

Reply:
Yes, good catch.